### PR TITLE
refactor: remove obsolete RuntimeInterface methods

### PR DESCRIPTION
**Pull request by alexjuda**
# The problem

It's been a while since we reflected on `RuntimeInterface` methods. Originally this interface was public so we couldn't remove old methods; we only kept adding new ones. Nowadays, this interface isn't used by our users, so we're free to change it how we like.

# This PR's solution

Removes methods we don't use/implement anymore.

- `RuntimeInterface.get_workflow_run_outputs()`
- `RuntimeInterface.get_all_workflow_runs_status()`. The only usage is replaced with `list_workflow_runs()`.

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`

### CODE REVIEW

File: src/orquestra/sdk/_ray/_dag.py [hunk 1]
```diff
@@ -862,7 +842,12 @@ def get_workflow_run_outputs_non_blocking(
                 f"{workflow_run_id} has not succeeded. {workflow_state}",
                 workflow_state,
             )
-        return self.get_workflow_run_outputs(workflow_run_id)
+        try:
+            return self._client.get_workflow_output(workflow_run_id)
+        except ValueError as e:
+            raise exceptions.NotFoundError(
+                f"Workflow run {workflow_run_id} wasn't found"
+            ) from e
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_ray/_dag.py:850]: **issue:** ‏These are uncovered, can you add a test to make sure this happens?

File: tests/runtime/ray/test_integration.py [hunk 2]
```diff
@@ -546,81 +547,89 @@ def test_run_and_get_output(
...
-
-    We test those 2 things together as package installation is long, and we don't
-    want to do that twice unnecessarily
-    """
-    # This package should not be installed before running test
-    with pytest.raises(ModuleNotFoundError):
-        import polars  # type: ignore # noqa
+class Test3rdPartyLibraries:
+    @staticmethod
+    def test_constants_n_inline_imports(runtime: _dag.RayRuntime):
```
Comment (jamesclark-Zapata) [inline tests/runtime/ray/test_integration.py:573]: **nitpick:** ‏Please update this test name, it looks incomplete.

Comment (alexjuda) [reply tests/runtime/ray/test_integration.py:573]: What name would make more sense to you?

File: src/orquestra/sdk/_ray/_dag.py (hunk 1 above)
Comment (alexjuda) [reply src/orquestra/sdk/_ray/_dag.py]: It was a case that's unlikely to happen in normal circumstances. I simplified it in bad2622

File: tests/runtime/ray/test_integration.py (hunk 2 above)
Comment (alexjuda) [reply tests/runtime/ray/test_integration.py]: Reworded in 34627d5

**Review by jamesclark-Zapata — APPROVED** [review]

**Pull request merged**